### DISCUSSION TITLE
docs/guide/composition/backend/

### DISCUSSION THREAD

**Discussion by giscus** [iPanelDev/docs#41] [Giscus]
# docs/guide/composition/backend/

Static Badge

https://ipaneldev.github.io/docs/guide/composition/backend/

**Comment by chenchen71956** [comment]
您好，我在后端-控制台发现有一个命令一直在跳：从广播WS服务器断开连接，但是我可以正常登录网页，至于能不能进行控制操作，我不知道，我没有创建用户。（从广播WS服务器断开连接前面的端口一直在变）

**Reply by Zaitonn** [reply]
刷新下浏览器，如果**登录后**还能够复现的话按一下`F12`截图一下控制台输出

>从广播WS服务器断开连接

理论上这个现象应该是正常且不影响使用的

>从广播WS服务器断开连接前面的端口一直在变

这个能麻烦详细说明一下吗

**Reply by chenchen71956** [reply]
抱歉，我无法再复现这个问题
此外，我还想知道登录时的md5生成逻辑，十分感谢！！！

**Reply by Zaitonn** [reply]
### 用户

https://ipaneldev.github.io/docs/development/api/users/login#md5%E8%AE%A1%E7%AE%97%E6%96%B9%E6%B3%95
https://github.com/iPanelDev/WebConsole/blob/e5801868d2d715a08cd6d0b27470885973a9591f/src/service/requests.ts#L24-L52

### 实例

https://ipaneldev.github.io/docs/development/webSocket/request/verify

**Reply by chenchen71956** [reply]
启动顺序是固定的，必须先启动iPanel再启动所有服务端，先重启服务端再启动iPanel只会尝试发送消息，但是不会再尝试连接。
我希望发送失败则尝试重新连接，这样即便我启动顺序错误也可以正常运行，只是会丢失一些信息。

**Reply by Zaitonn** [reply]
没太看懂是什么意思🤔

你用的是mcdr吗

**Reply by chenchen71956** [reply]
是的，就是我必须先启动 iPanel.exe 再启动 run.bat，如果顺序错了就不会再次尝试连接了。
也就是 mcdr 的插件只会在启动时尝试连接一次，我希望他可以在向 iPanel 发送失败时尝试重新连接，这样就可以不用管启动顺序啦😊

**Reply by Zaitonn** [reply]
了解

但是最近没有更新的计划:(  
如果有能力的话可以自己修改源码

**Reply by chenchen71956** [reply]
okok，当然有能力😘

**Reply by Zaitonn** [reply]
https://github.com/iPanelDev/iPanel-MCDReforged/blob/main/ipanel_mcdreforged/__init__.py#L19-L26

改一下加个循环判断即可

**Comment by chenchen71956** [comment]
我尝试把这个插件删除重新配置，虽然WS广播还在弹，但是我能使用了
